src/components/LeadsTable/leadsSlice.ts

```diff
@@ -0,0 +1,119 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
```

Comment:
**issue:** Data layer should not be a part of UI layer. You need to split it, and create other directory. Eg. `src/api/lead/leadSlice.ts`. Think of it like your data layer should be totally independent from UI. If you'd like to reuse it in other project using Vue.js for example, you should be possible to do that